src/Uno.UI.Runtime.Skia.Android/AndroidSkiaTextBoxNotificationsProviderSingleton.cs

```diff
@@ -32,17 +32,28 @@ public void OnUnfocused(TextBox textBox)
 	{
 		if (UnoSKCanvasView.Instance is { } canvasView)
 		{
-			// Hide the keyboard only when the next focused element is not a TextBox
+			// Hide the keyboard only when the next element to be focused is not an Element that
+			// could require the keyboard (TextBox, AutoSuggestBox, NumberBox, etc.).
 			// This prevents the keyboard from flickering when switching between TextBoxes
 			// https://github.com/unoplatform/uno-private/issues/1160
-			var xamlRoot = textBox.XamlRoot;
-			if (xamlRoot is null || FocusManager.GetFocusedElement(xamlRoot) is not TextBox)
+			if (IsFocusingElementKeyboardActivator(textBox.XamlRoot))
 			{
 				canvasView.TextInputPlugin.HideTextInput();
 			}
 
 			canvasView.TextInputPlugin.NotifyViewExited(textBox.GetHashCode());
 		}
+
+		static bool IsFocusingElementKeyboardActivator(XamlRoot? xamlRoot)
+		{
+			if (xamlRoot is null)
+			{
+				return true;
+			}
+
+			var focusingElement = FocusManager.GetFocusingElement(xamlRoot) as FrameworkElement;
+			return !CouldRequireKeyboard(focusingElement);
```

Comment:
Is this condition right? It feels like `IsFocusingElementKeyboardActivator` should return `true` if the target element `CouldRequireKeyboard` 🤔

Reply:
I see, the condition is correct, it's just that the `!` is in the wrong spot. I should not be negating the `CouldRequireKeyboard`, and as you say, this should return true. Instead, the `!` should be on line 30 when we call the method `IsFocusingElementKeyboardActivator`

```
Hide the keyboard only when the next element to be focused is not an Element that
could require the keyboard (TextBox, AutoSuggestBox, NumberBox, etc.).
```

Fixing now and thanks for the comment
cc: @MartinZikmund